models/entry.go

```diff
@@ -57,3 +57,17 @@ func (e *Entry) AddStringValue(value string, units string) {
 	val.Units = units
 	e.Values = append(e.Values, val)
 }
+
+func (e *Entry) PreferredCode(preferredCodeSets []string) Concept {
```

Comment:
This method does not take value sets into account.  You should review what is in HDS to see how they are used in that method